### DISCUSSION TITLE
beIN Sports Connect streams?

### DISCUSSION THREAD

**Discussion by btgio** [streamlink/streamlink#3956] [Q&A]
Hi there.

I am a novice with this stuff but I figured out how to work streamlink with ESPN+. I am trying to do the same with beIN Sports Connect for Messi's games but it is proving more difficult. 

I don't get any handy m3u8 links and instead am stuck with this

https://dvr-d-main-beinusca.akamaized.net/dvr/6b39fd6b-2d74-4c09-8f95-4a99171d8649/6b39fd6b-2d74-4c09-8f95-4a99171d8649.ism/manifest(format=mpd-time-csf,filter=desktop-fullres,encryption=cenc)?hdnts=st=1630184407~exp=1630184437~acl=/dvr/6b39fd6b-2d74-4c09-8f95-4a99171d8649/*~data=30c5a27c-3fc3-4820-af54-87e385b4ebfa~hmac=a9a0a735aa2a58f5c75c9f31a5423053e6997469a72dbce0a0fc672b31dd8d8a&hdcore=2.11.3

I've tried multiple interactions of a streamlink command but nothing works. The closest thing to success I get is the line:

Available streams: live (worst, best)

Which never resolves itself.

Does anyone know how to pull the full matches from beIN's DVR collection?

**Comment by mkbloke** [comment]
It might possibly help to specify `Origin` and `Referer` headers.  That's all I can suggest, as it's not possible to investigate further due to the login requirement on that site.

See: https://github.com/streamlink/streamlink/discussions/3912#discussioncomment-1161074 for an example of specifying those headers.
